### ISSUE TITLE
Error installing basehash

### ISSUE THREAD

**Issue by dustin999** [bnlucas/python-basehash#6]
Getting the following in windows linux subshell (inside a python 3.6.9 virtualenv)

```
$ pip install basehash==3.0.4 --verbose
Created temporary directory: /tmp/pip-ephem-wheel-cache-aenp6ail
Created temporary directory: /tmp/pip-req-tracker-h9d5w_fc
Created requirements tracker '/tmp/pip-req-tracker-h9d5w_fc'
Created temporary directory: /tmp/pip-install-_v90amq4
Collecting basehash==3.0.4
  1 location(s) to search for versions of basehash:
  * https://pypi.org/simple/basehash/
  Getting page https://pypi.org/simple/basehash/
  Found index url https://pypi.org/simple
  Looking up "https://pypi.org/simple/basehash/" in the cache
  Request header has "max_age" as 0, cache bypassed
  Starting new HTTPS connection (1): pypi.org:443
  https://pypi.org:443 "GET /simple/basehash/ HTTP/1.1" 304 0
  Analyzing links from page https://pypi.org/simple/basehash/
    Found link https://files.pythonhosted.org/packages/9b/06/35e20d0937ad1c9a2e9f168b8ba0756b51197790c709c4cc111d4520f1d4/BaseHash-1.0.5.zip#sha256=6277e289d93d271044cdeb8231d578c9498278dfd82a87a19eba9eaa535cbc08 (from https://pypi.org/simple/basehash/), version: 1.0.5
    Found link https://files.pythonhosted.org/packages/ce/15/99b5239e086eb21d00e3a1fb42d3ee521503e76ac24f9e6875c565b30033/BaseHash-1.0.6.zip#sha256=139c076667224bd2616787e9d2caab6856722f9cc1e56af6c4f54a008d2d3d40 (from https://pypi.org/simple/basehash/), version: 1.0.6
    Found link https://files.pythonhosted.org/packages/41/70/643bf8502682da07c68b5c214f8ac3edcdf2464559c3c61e9e5b135338f3/BaseHash-1.0.7.zip#sha256=de077fc8bfc816c8a31c453ae08b4f263a6edd8aa6616b3e8be74970e3db029c (from https://pypi.org/simple/basehash/), version: 1.0.7
    Found link https://files.pythonhosted.org/packages/91/7e/38a908e1e03e719835dffb303fc6ec49bb24e055ca3b02dbdf912427cc46/BaseHash-1.0.7.1.zip#sha256=9f128804d513d6423571905009a6b4762923b7300f399672263bf292534375d9 (from https://pypi.org/simple/basehash/), version: 1.0.7.1
    Found link https://files.pythonhosted.org/packages/79/0c/50e6df0422000992af5f9143e10b1271203567214badb3bd80876eb2cc15/BaseHash-2.0.0.zip#sha256=cf8bb987088d4eec6b2c0beb9a2bef5d2f503bc37db37df945d797311358412a (from https://pypi.org/simple/basehash/), version: 2.0.0
    Found link https://files.pythonhosted.org/packages/1a/bb/0bf626c9a1da49c6e13b76163befe121086663a056bdd06000ea8ad2666a/BaseHash-2.0.2.zip#sha256=5e66d21f856a3d2c8f7984ffdb639ab41769d563a22dd377f4c33fd1e5954648 (from https://pypi.org/simple/basehash/), version: 2.0.2
    Found link https://files.pythonhosted.org/packages/7a/92/d982c48cf41bc15b61b12323638601de11941a3f15fb6002050c94ba8251/BaseHash-2.1.0.zip#sha256=ecd31c13b47c50df54bb05f3ddaafc171cd4f72a5732d98241283028f79d27b5 (from https://pypi.org/simple/basehash/), version: 2.1.0
    Found link https://files.pythonhosted.org/packages/bd/bd/ebb635630498363c5c3caaf4f90e84f7fe1b5382f02311cb1bacfba4d300/BaseHash-2.2.1.zip#sha256=0e64d1b33d424cab970cc8b92ff1cc8bfd76de7b841e0de375993c9d3e9bafaa (from https://pypi.org/simple/basehash/), version: 2.2.1
    Found link https://files.pythonhosted.org/packages/38/26/485b661654ef934616665a8ee12ea77ff286bc013211e0fa8c4c1b29de94/BaseHash-3.0.0.zip#sha256=a1548a54650d15c5b0721ce7bbf89dbdeedee817a3d21297805c163dd61debfe (from https://pypi.org/simple/basehash/), version: 3.0.0
    Found link https://files.pythonhosted.org/packages/f7/05/f1959c881d7cf40c77553ffc5aafcfbd4ddee94d5f93230a358cae4522fd/BaseHash-3.0.1.zip#sha256=15c0f4377aeddd74424c361fedf544cf0ee44164db788a3fdf834ca1d918be8e (from https://pypi.org/simple/basehash/), version: 3.0.1
    Found link https://files.pythonhosted.org/packages/fa/5b/9761a8a0ae3dc80dc3565282d40407761d6c0f30b650e173a661e06db532/BaseHash-3.0.4.zip#sha256=1da4f16e6b760bc00b37e544452d59659be5ab350a68ec35ccb542f6fee811e8 (from https://pypi.org/simple/basehash/), version: 3.0.4
  Given no hashes to check 1 links for project 'basehash': discarding no candidates
  Using version 3.0.4 (newest of versions: 3.0.4)
  Created temporary directory: /tmp/pip-unpack-141iij0p
  Looking up "https://files.pythonhosted.org/packages/fa/5b/9761a8a0ae3dc80dc3565282d40407761d6c0f30b650e173a661e06db532/BaseHash-3.0.4.zip" in the cache
  Current age based on date: 3497
  Ignoring unknown cache-control directive: immutable
  Freshness lifetime from max-age: 365000000
  The response is "fresh", returning cached response
  365000000 > 3497
  Using cached https://files.pythonhosted.org/packages/fa/5b/9761a8a0ae3dc80dc3565282d40407761d6c0f30b650e173a661e06db532/BaseHash-3.0.4.zip
  Downloading from URL https://files.pythonhosted.org/packages/fa/5b/9761a8a0ae3dc80dc3565282d40407761d6c0f30b650e173a661e06db532/BaseHash-3.0.4.zip#sha256=1da4f16e6b760bc00b37e544452d59659be5ab350a68ec35ccb542f6fee811e8 (from https://pypi.org/simple/basehash/)
  Added basehash==3.0.4 from https://files.pythonhosted.org/packages/fa/5b/9761a8a0ae3dc80dc3565282d40407761d6c0f30b650e173a661e06db532/BaseHash-3.0.4.zip#sha256=1da4f16e6b760bc00b37e544452d59659be5ab350a68ec35ccb542f6fee811e8 to build tracker '/tmp/pip-req-tracker-h9d5w_fc'
    Running setup.py (path:/tmp/pip-install-_v90amq4/basehash/setup.py) egg_info for package basehash
    Running command python setup.py egg_info
    error in BaseHash setup command: "values of 'package_data' dict" must be a list of strings (got '*.py')
Cleaning up...
  Removing source in /tmp/pip-install-_v90amq4/basehash
Removed basehash==3.0.4 from https://files.pythonhosted.org/packages/fa/5b/9761a8a0ae3dc80dc3565282d40407761d6c0f30b650e173a661e06db532/BaseHash-3.0.4.zip#sha256=1da4f16e6b760bc00b37e544452d59659be5ab350a68ec35ccb542f6fee811e8 from build tracker '/tmp/pip-req-tracker-h9d5w_fc'
Removed build tracker '/tmp/pip-req-tracker-h9d5w_fc'
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
Exception information:
Traceback (most recent call last):
  File "/home/dustin/.virtualenvs/designhub/lib/python3.6/site-packages/pip/_internal/cli/base_command.py", line 188, in main
    status = self.run(options, args)
  File "/home/dustin/.virtualenvs/designhub/lib/python3.6/site-packages/pip/_internal/commands/install.py", line 345, in run
    resolver.resolve(requirement_set)
  File "/home/dustin/.virtualenvs/designhub/lib/python3.6/site-packages/pip/_internal/legacy_resolve.py", line 196, in resolve
    self._resolve_one(requirement_set, req)
  File "/home/dustin/.virtualenvs/designhub/lib/python3.6/site-packages/pip/_internal/legacy_resolve.py", line 359, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "/home/dustin/.virtualenvs/designhub/lib/python3.6/site-packages/pip/_internal/legacy_resolve.py", line 307, in _get_abstract_dist_for
    self.require_hashes
  File "/home/dustin/.virtualenvs/designhub/lib/python3.6/site-packages/pip/_internal/operations/prepare.py", line 215, in prepare_linked_requirement
    finder, self.build_isolation,
  File "/home/dustin/.virtualenvs/designhub/lib/python3.6/site-packages/pip/_internal/distributions/source.py", line 79, in prepare_distribution_metadata
    self.req.prepare_metadata()
  File "/home/dustin/.virtualenvs/designhub/lib/python3.6/site-packages/pip/_internal/req/req_install.py", line 551, in prepare_metadata
    self.run_egg_info()
  File "/home/dustin/.virtualenvs/designhub/lib/python3.6/site-packages/pip/_internal/req/req_install.py", line 628, in run_egg_info
    command_desc='python setup.py egg_info')
  File "/home/dustin/.virtualenvs/designhub/lib/python3.6/site-packages/pip/_internal/utils/misc.py", line 931, in call_subprocess
    raise InstallationError(exc_msg)
pip._internal.exceptions.InstallationError: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by colinmcdonald22** [comment]
I'm having the same problem on Python 3.7.4. Downgrading to 3.7.3 appears to work around the issue.

**Comment by bnlucas** [comment]
I've updated `setup.py` but was never able to replicate the issue. I've added more versions to the travis-ci config and they are all passing.

Can someone confirm if the update is working for them?

**Comment by h1nk** [comment]
Can confirm latest master branch changes seem to fix the issues for 3.7.4 etc.

**Comment by alxgb** [comment]
Researched it a bit, the problem comes not from Python 3.7.x but from the most recent setuptools version. Indeed that should always be an array

The change resulted from this issue: pypa/setuptools#1459
